pkg/sif/update.go

```diff
@@ -470,3 +490,21 @@ func removeRefAnnotation(ii v1.ImageIndex, ref name.Reference) (v1.ImageIndex, e
 	ii = mutate.RemoveManifests(ii, m)
 	return mutate.AppendManifests(ii, ia), nil
 }
+
+// RemoveBlob removes a blob from the SIF f, without modifying the rootIndex.
+func (f *OCIFileImage) RemoveBlob(hash v1.Hash) error {
+	return f.sif.DeleteObjects(sif.WithOCIBlobDigest(hash),
+		sif.OptDeleteZero(true),
+		sif.OptDeleteCompact(true))
+}
+
+// RemoveDescriptors modifies the SIF file associated with f so that its
+// RootIndex no longer holds descriptors selected by matcher. Any blobs in the
+// SIF that are no longer referenced are removed from the SIF.
+func (f *OCIFileImage) RemoveDescriptors(matcher match.Matcher) error {
```

Comment:
"Descriptors" might be a little confusing here? I believe this only removes matching _manifests_, rather than anything with a descriptor (ex. layers)?

Maybe `RemoveManifests`?

```suggestion
// RemoveManifests modifies the SIF file associated with f so that its
// RootIndex no longer references manifests selected by matcher. Any blobs in the
// SIF that are no longer referenced are removed from the SIF.
func (f *OCIFileImage) RemoveManifests(matcher match.Matcher) error {
```

Reply:
I've generally been approaching this thinking generally that the way to be least confusing is to be as close to ggcr `layout.Path` as possible... so calling code is using the same function names for the same operations.

```
// RemoveDescriptors removes any descriptors that match the match.Matcher from the index.json of the Path.
func (l Path) RemoveDescriptors(matcher match.Matcher) error {
```

Not opposed to renaming though.

Reply:
Yeah I think `RemoveManifests` is probably clearer... `(Path).RemoveDescriptors` is a lower level interface... it removes _only_ the descriptor from the root `index.json`... the caller also needs to call `(Path).RemoveBlob` on each manifest/config/layer to get equivalent behaviour to what you have here.